src/DynamoCore/DSEngine/AstBuilder.cs

```diff
@@ -31,6 +31,16 @@ public interface IAstNodeContainer
     /// </summary>
     public class AstBuilder : LogSourceBase
     {
+        /// <summary>
+        /// The context of AST compilation
+        /// </summary>
+        public enum CompilationContext
+        {
+            ForDeltaExecution,
+            ForNodeToCode,
+            None
+        }
+
```

Comment:
I'd recommend removal of `For` in the constant names, a prime example of this is the platform [`FileAccess` enumeration](https://msdn.microsoft.com/en-us/library/4z36sx0f(v=vs.110).aspx), it has `Read`, `Write` eventhough the flags meant `ForRead` and `ForWrite`, respectively.

Also, shouldn't `None` be the first entry? :)